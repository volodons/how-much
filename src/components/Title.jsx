import { StyledTypography } from '../styled/styledTitle';

const Title = () => <StyledTypography variant="h1">How Much?</StyledTypography>;

export default Title;
