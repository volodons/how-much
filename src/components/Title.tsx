import React from 'react';

import { StyledTypography } from '../styled/styledTitle';

const Title: React.FC = () => (
    <StyledTypography variant="h1" data-testid="title-component">
        How Much?
    </StyledTypography>
);

export default Title;
