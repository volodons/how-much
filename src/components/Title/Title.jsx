import Typography from "@mui/material/Typography";
import { styled } from "@mui/material";
import "typeface-lobster";

const StyledTypography = styled(Typography)`
    margin: 1rem;
    background-color: #0390fc;
    background: linear-gradient(to right, #0390fc 0%, #8c0dfc 100%);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    font-family: lobster, sans-serif;
    font-size: 5rem;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.3);
`;

const Title = () => {
    return <StyledTypography variant="h1">How Much?</StyledTypography>;
};

export default Title;
