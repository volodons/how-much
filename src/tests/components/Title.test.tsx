import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Title from '../../components/Title';

describe('Title component', () => {
    it('should be a functional component', () => {
        expect(Title).toBeInstanceOf(Function);
    });

    it('should render with correct tag', () => {
        render(<Title />);
        const titleElement = screen.getByTestId('title-component');
        expect(titleElement.tagName).toBe('H1');
    });

    it('should display correct text', () => {
        render(<Title />);
        const titleElement = screen.getByTestId('title-component');
        expect(titleElement.textContent).toBe('How Much?');
    });

    it('should match a snapshot', () => {
        const tree = renderer.create(<Title />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
