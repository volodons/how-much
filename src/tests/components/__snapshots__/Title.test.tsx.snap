// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should match a snapshot 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1 css-1dha1by-MuiTypography-root"
  data-testid="title-component"
>
  How Much?
</h1>
`;
